File: ClientApp/src/components/Assignment.js
import {formatDateToMonthAndYear} from "../FormatDate";
import "./Assignment.css";

const Assignment = ({assignment}) => {
    return <div>
        {assignment.name} ({formatDateToMonthAndYear(assignment.startDate)} - {formatDateToMonthAndYear(assignment.endDate)})
        <div className="assignment-summary">
            {assignment.description}
        </div>
        <div className="duties">

            {assignment.duties.map((duty) => [
                <span>   {duty} <br/></span>
            ])}
        </div>

        <div className="assignment-summary">
            Tech:&nbsp;
            {assignment.technologies.map((tech, i) => [
                i > 0 && ", ",
                <span>{tech}</span>
            ])}
        </div>

        {assignment.link && <div><a href={assignment.link.address} target="_blank"> {assignment.link.caption || <span>See Details</span>}</a></div>}
    </div>
}

export default Assignment;

File: Employments/Records/Assignment.cs
using System;
using System.Collections.Generic;
using BusinessCard.Technologies.Records;

namespace BusinessCard.Employments.Records
{
    public class Assignment
    {
        public int Id { get; set; }

        public string Role { get; set; }

        public string Name { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public virtual Employment Employment { get; set; }

        public virtual ICollection<Duty> Duties { get; set; } = new List<Duty>();
        public string Description { get; set; }
        public string Summary { get; set; }
        public virtual ICollection<Technology> Technologies { get; set; } = new List<Technology>();

        public virtual AssignmentLink? Link { get; set; }
    }

    public class AssignmentLink
    {
        public string Address { get; set; }

        public string Caption { get; set; }
    }
}